types/index.d.ts

```diff
@@ -11,8 +10,8 @@ declare namespace envSchema {
     [key: string]: unknown;
   };
 
-  export type EnvSchemaOpt<T = EnvSchemaData> = {
-    schema?: JSONSchemaType<T> | AnySchema;
+  export type EnvSchemaOpt = {
+    schema?: object;
```

Comment:
Did you evaluate adding the fluent-schema's `JSONSchema` type?

Reply:
I am not quite sure what you mean by that. Are you suggesting to add `JSONSchema` type from `fluent-json-schema` to union type for schema? If so, read description of this pull request

Reply:
Hmm. object is not a very good type. Should use something like Record<string, any> or so.

Reply:
And to be honest i dont understand why  we remove the generic type

Reply:
Because it is not compatible to different JSON schema library.

Reply:
I think you should read backstory of this pull request - #134 , #137 , #138 . I think it will become clear to you

As for `typescript`:
`Record<string, any>` and `object` are equal. Also, we can not use `Record<string, unknown>` type, because it will not work with interfaces ([typescript playground](https://www.typescriptlang.org/play?ts=4.9.5#code/JYOwLgpgTgZghgYwgAgJIEZkG8CwAoZZGAe2IC5kBnMKUAcwG58BfffMATwAcUAVTALzZ8hEuSo16TPM2n4ExENWQA3dBQzIhuAkVIUA5GIMAaFnLwKlYVQCYK-LcN1jDxszItXlMEAHkAIwArCAQbIQAKOCg6CmJg0LAASi0APmxZNktFHxAAJVDiKAATAEEQDicomIoChRKAHmpaEDoTZDgK1JSBdKxMvHkcm186ouKAVRAAaxBiAHcQKujY5DHG5vp2gFcZucXutIyLX0CQsIi1JOlThIuVW2us0cKS8o5L9Ce8F-qyisujxO+Vekz2CxAn2+v3GU1mEMB3yG1iocBgEF4EGo6mQ8XO4WQAG0ALrSbw2ShojFYsD2NagpqSVrtTocdJCElk4ao9GY6gAZlqDM2zOQu3hBycnPwQA))